pyiceberg/table/__init__.py

```diff
@@ -2487,6 +2491,7 @@ def __init__(
         operation: Operation,
         transaction: Transaction,
         io: FileIO,
+        snapshot_properties: Dict[str, str],
         commit_uuid: Optional[uuid.UUID] = None,
```

Comment:
Suggestion to avoid breaking the positional order:
```suggestion
        commit_uuid: Optional[uuid.UUID] = None,
        snapshot_properties: Dict[str, str] = EMPTY_DICT,
```